packages/middleware/src/entries.js

```diff
@@ -10,6 +10,15 @@ _ENTRIES['middleware_pages/_middleware'] = {
         response: result,
       };
     }
-    return result;
+
+    return {
+      promise: Promise.resolve(),
+      waitUntil: Promise.resolve(),
+      response: new Response(null, {
```

Comment:
@AndyBitz is this assumption correct? Whenever middleware returns something that isn't a `Response` we essentially call next?

Reply:
Yes, but we should check what Next.js does to make sure we don't create any differences in that behaviour

Reply:
@AndyBitz I checked next.js and here's what they're doing - https://github.com/vercel/next.js/blob/canary/packages/next/server/web/adapter.ts#L35

No checking for the result to be an instance of Response, if it's truthy it's returned. I've updated this PR to do the same.